src/logic/embedding.lean

```diff
@@ -34,6 +34,16 @@ end function
 protected def equiv.to_embedding {α : Sort u} {β : Sort v} (f : α ≃ β) : α ↪ β :=
 ⟨f, f.injective⟩
 
+/-- Given an equivalence to a coerced set `↥s`, produce an embedding to the elements of that set. -/
+@[simps]
+def equiv.as_embedding {α β : Sort*} {s : set β} (e : α ≃ ↥s) : α ↪ β :=
```

Comment:
This name is a little similar to `equiv.to_embedding`.

Also, perhaps we should generalize `↥s` to `subtype p`.

Reply:
That is,
```
/-- Given an equivalence to a subtype, produce an embedding to the elements of enclosing type. -/
@[simps]
def equiv.embedding_of_subtype {α β : Type*} {p : β → Prop} (e : α ≃ subtype p) : α ↪ β :=
e.to_embedding.trans $ function.embedding.subtype p
```

Reply:
I know the name is similar, but I don't have a better idea. `equiv.to_embedding` has the right to use this name, and I didn't want to use a very long name. Suggestions are welcome.

Reply:
You're not a fan of the name `embedding_of_subtype` then?

Reply:
It should be `embedding_of_equiv_subtype` then, right? I think `equiv.as_embbeding` is fine.
Also, sorry for kicking this on the queue. I missed that this conversation hadn't come to a conclusion yet.

Reply:
I figured the word `equiv` was redundant, hence dropping it from `embedding_of_equiv_subtype`.

Getting `subtype` in the name will probably make this more discoverable.